[DOC] Add CDAT API mapping table and gallery examples (#239)
xCDAT/xcdat: README.rst

> Initial features include:

Author: ```suggestion
Initial planned features include:
```

* Extension of xarray's ``open_dataset()`` and ``open_mfdataset()`` with post-processing options

  * Generate bounds for axes supported by ``xcdat`` if they don't exist in the Dataset
  * Optional decoding non-CF time units, in addition to CF time units (already supported in ``xarray``)
  * Optional centering of time coordinates using time bounds
  * Optional conversion of longitudinal axis orientation between [0, 360) and [-180, 180)

* Temporal averaging

  * Time series averages (single snapshot and grouped), climatologies, and departures
  * Weighted or unweighted

>   * Optional seasonal configuration

Author: ```suggestion
  * Optional seasonal configuration (e.g., DJF vs. JFD, define custom seasons)
```

Initial features include:

* Extension of xarray's ``open_dataset()`` and ``open_mfdataset()`` with post-processing options

  * Generate bounds for axes supported by ``xcdat`` if they don't exist in the Dataset
  * Optional decoding non-CF time units, in addition to CF time units (already supported in ``xarray``)
  * Optional centering of time coordinate using time bounds
  * Optional conversion of longitudinal axis orientation between [0, 360) and [-180, 180)

* Temporal averaging

>   * Time series averages, climatologies, and departures

Author: ```suggestion
  * Time series averages (single snapshot and grouped), climatologies, and departures
```

  * Weighted or unweighted

>   * Optional centering of time coordinates using time bounds

Author: ```suggestion
```

  * Optional seasonal configuration

* Geospatial weighted averaging

  * Support rectilinear grid
  * Optional specification of regional domain

* Horizontal structured regridding

  * Support rectilinear and cuvilinear grids
  * Python implementation of `regrid2`_ for handling cartesian latitude longitude grids

>   * Utilize `xesmf`_ with utilities to handle edge cases

Author: ```suggestion
  * API that wraps `xesmf`_ with utilities to handle edge cases
```

* Vertical structured regridding

  * Support rectilinear and cuvilinear grids

> Things we keep in mind:

Author: ```suggestion
Things we are striving for:
```

* Support for CF compliant, E3SM non-CF compliant, and common metadata

  * Leverage `cf_xarray`_ to interpret `CF convention`_ attributes on ``xarray`` objects

* Robust handling of coordinates and its associated bounds

>   * Coordinates and bounds are retrieved with ``cf_xarray`` using the standard ``axis`` and ``coordinate`` attributes

Author: ```suggestion
  * Coordinates and bounds are retrieved with ``cf_xarray`` using the standard ``axis`` and ``coordinate`` name attributes
```
